tests/lean/split-ifs.lean

```diff
@@ -0,0 +1,21 @@
+/-! Test that the `split` tactic doesn't use `Classical.em` to split ifs.
+-/
+
+/- Basic example -/
+theorem ittt1 (n : Nat) : (if n = 0 then 0 else 0) = 0 := by split <;> rfl
+
+#print axioms ittt1
+
+/- Example where the decidabilty instance has loose bound variables
+  and split falls back on classical axioms (courtesy of Mario Carneiro). -/
+theorem ittt2 (c : Prop) :
+    (fun [Decidable c] => if c then 0 else 0) = (fun _ => 0) := by
+  split <;> rfl
+
+#print axioms ittt2
```

Comment:
Just to note, this is consistent with the Lean 3 behavior:
```lean
import tactic.split_ifs

theorem ittt2 (c : Prop) :
  (fun [decidable c], by exactI if c then 0 else 0) = (fun _, 0) :=
begin
  split_ifs,
  sorry,
  sorry
end

#print axioms ittt2
```
(suggesting the choice that was made to fallback to classical rather than failing is the correct one)